Source/WebCore/Modules/mediasource/SourceBuffer.cpp

```diff
@@ -1917,19 +1917,6 @@ void SourceBuffer::sourceBufferPrivateDidReceiveSample(MediaSample& sample)
     // duration set to the maximum of the current duration and the group end timestamp.
     if (m_groupEndTimestamp > m_source->duration())
         m_source->setDurationInternal(m_groupEndTimestamp);
-
```

Comment:
You can keep this code valid by referring to the new fields, like this:

<pre>
    // To avoid playback pipeline starvation start providing media data as soon as we can
    const auto& trackID = sample.trackID();
    auto it = m_trackBufferMap.find(trackID);
    if (it != m_trackBufferMap.end() && m_private->isReadyForMoreSamples(trackID)) {
        TrackBuffer& trackBuffer = it->value;

        if (!trackBuffer.needsReenqueueing
            && trackBuffer.lastEnqueuedDecodeKey.first.isValid()
            && trackBuffer.lastEnqueuedDecodeDuration.isValid()
            && trackBuffer.lastDecodeTimestamp.isValid()) {
            MediaTime lastEnqueuedDecodeEndTime = trackBuffer.lastEnqueuedDecodeKey.first + trackBuffer.lastEnqueuedDecodeDuration;
            if (abs(trackBuffer.lastEnqueuedDecodeKey.first + trackBuffer.lastEnqueuedDecodeDuration - trackBuffer.lastDecodeTimestamp) > MediaTime(350, 1000))
                provideMediaData(trackBuffer, trackID);
        }
    }
</pre>

However, as this block of code was originally added by you in https://github.com/WebPlatformForEmbedded/WPEWebKit/commit/06b35a846bcb8cf9d183a32b3e7119bd5c6d9250 , it's up to you to decide if you want to remove it or keep it.

Reply:
Hi Enrique, thanks for the review.

AppendPipeline changed a lot since the original change, and this doesn't seem to be required anymore. I'd prefer to remove it.

Also, we later found that in some cases SB appends were still taking a lot of time (5-10 seconds and more). That was caused by some timer source dispatched on the main loop, even though GST bus had higher priority. Turned out that GLib doesn't dispatch fd sources when there is a timer source ready.  I filed a ticket with a fix here: https://gitlab.gnome.org/GNOME/glib/-/issues/1592